src/ncp-spinel/SpinelNCPTaskJoin.cpp

```diff
@@ -139,11 +139,15 @@ nl::wpantund::SpinelNCPTaskJoin::vprocess_event(int event, va_list args)
 			goto on_error;
 		}
 
-		mNextCommand = SpinelPackData(
-			SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_BOOL_S),
-			SPINEL_PROP_THREAD_ROUTER_ROLE_ENABLED,
-			router_role_enabled
-		);
+		// only set router role (default disabled) if required.
```

Comment:
`wpantund` was not originally designed/intended to work with MTD NCP build. So I guess we may need more changes to make it fully compatible. 

I am fine with this change... but I think later we should introduce an `SPINEL_CAP_<>` to indicate if NCP is running MTD or FTD build and behave differently in `wpantund`.

Reply:
agree with the proposal.